src/uart_comm_unix.c

```diff
@@ -360,6 +360,92 @@ static int uart_config_flowcontrol(int fd, const struct uart_config *config)
     return tcsetattr(fd, TCSANOW, &options);
 }
 
+int uart_get_rs485_config(struct uart *port, struct uart_config *config)
+{
+#if defined(__linux__)
+    struct serial_rs485 rs485;
+
+    if (ioctl(port->fd, TIOCGRS485, &rs485) < 0) {
+        // This may fail with ENOTTY if the port doesn't support RS485. If thats
+        // the case, then we only care about a failure when a user is trying
+        // to set things. Otherwise, we assume RS485 may not even be expected and
+        // can safely be ignored
+        if (!config->rs485_user_configured) return 0;
+
+        record_errno();
+        return -1;
+    }
+
+    config->rs485_enabled = (rs485.flags & SER_RS485_ENABLED);
+    config->rs485_rts_on_send = (rs485.flags & SER_RS485_RTS_ON_SEND);
+    config->rs485_rts_after_send = (rs485.flags & SER_RS485_RTS_AFTER_SEND);
+    config->rs485_rx_during_tx = (rs485.flags & SER_RS485_RX_DURING_TX);
+    config->rs485_terminate_bus = (rs485.flags & SER_RS485_TERMINATE_BUS);
+
+    rs485.delay_rts_before_send = config->rs485_delay_rts_before_send >= 0 ? config->rs485_delay_rts_before_send : rs485.delay_rts_before_send;
+    rs485.delay_rts_after_send = config->rs485_delay_rts_after_send >= 0 ? config->rs485_delay_rts_after_send : rs485.delay_rts_after_send;
+#endif
+
+    return 0;
+}
+
+/**
+ * @brief Update the tristate flag of RS485 configution
+ *
+ * @param flags
+ * @param current_val
+ * @param rs485 setting mask
+ */
+static void update_tristate_flags(long *flags, int val, long mask)
+{
+    switch (val){
+    case 0: *flags |= mask; break;
+    case 1: *flags &= ~mask; break;
+    default: break;
+    }
```

Comment:
```suggestion
    if (val == 0)
        *flags |= mask;
    else if (val > 0)
       *flags &= ~mask;
```

^--- updated to be less picky on boolean true being exactly `1`